src/main/java/com/azure/cosmos/kafka/connect/sink/BulkWriter.java

```diff
@@ -56,15 +54,38 @@ protected SinkWriteResponse writeCore(List<SinkRecord> sinkRecords) {
         }
 
         List<CosmosItemOperation> itemOperations = new ArrayList<>();
-        for (SinkRecord sinkRecord : sinkRecords) {
-            CosmosItemOperation cosmosItemOperation = CosmosBulkOperations.getUpsertItemOperation(
-                    sinkRecord.value(),
-                    this.getPartitionKeyValue(sinkRecord.value()),
-                    new SinkOperationContext(sinkRecord));
+        if (this.noDuplicates) {
+            HashMap<IdAndPartionKey, SinkRecord> uniqueItems = new HashMap<>();
+            for (SinkRecord sinkRecord : sinkRecords) {
+                IdAndPartionKey idAndPartionKey = new IdAndPartionKey(((Map<String, String>) sinkRecord.value()).get("id"), this.getPartitionKeyValue(sinkRecord.value()));
```

Comment:
((Map<String, String>) sinkRecord.value()) -> shouldn't this be `Map<String, Object>`?

Reply:
Yes it should be.